server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CachedBlobStore.java

```diff
@@ -138,17 +138,19 @@ public InputStream read(BucketName bucketName, BlobId blobId, StoragePolicy stor
         if (storagePolicy == LOW_COST) {
             return backend.read(bucketName, blobId);
         }
-        return Mono.just(bucketName)
-            .filter(getDefaultBucketName()::equals)
-            .flatMap(defaultBucket -> readInDefaultBucket(bucketName, blobId))
-            .switchIfEmpty(readFromBackend(bucketName, blobId))
+        if (bucketName.equals(getDefaultBucketName())) {
```

Comment:
Well as a rule of thumb simplifying the reactor execution lead to less big call stacks, and less runtime steps.

IMO we should refrain from pushing unneeded complexity to reactor.

Reply:
I guess I am put off by the repetition of the error handling block, it's bigger than the code it protects and makes the difference between the 2 conditional  branches harder to spot. 
```java
.blockOptional()
.orElseThrow(() -> new ObjectNotFoundException(String.format("Could not retrieve blob metadata for %s", blobId.asString())));
```
defining 
```java
private Mono<InputStream> readFromBucket(String bucketName){
 if (bucketName.equals(getDefaultBucketName())) {
     return readInDefaultBucket(bucketName, blobId)
  } else {
     return readFromBackend(bucketName, blobId)
  }
}
```
maybe ?

if I understand correctly the special casing of `readInDefaultBucket` is done only because there is caching on the default bucket ...